packages/lodestar/src/chain/validation/syncCommittee.ts

```diff
@@ -0,0 +1,94 @@
+import {CachedBeaconState, computeSubnetsForSyncCommittee, fast} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {altair} from "@chainsafe/lodestar-types";
+import {IBeaconDb} from "../../db";
+import {ISyncCommitteeJob, SyncCommitteeError, SyncCommitteeErrorCode} from "../errors/syncCommitteeError";
+import {IBeaconChain} from "../interface";
+
+export async function validateGossipSyncCommittee(
+  config: IBeaconConfig,
+  chain: IBeaconChain,
+  db: IBeaconDb,
+  syncCommitteeJob: ISyncCommitteeJob,
+  subnet: number
+): Promise<void> {
+  const {syncCommittee, validSignature} = syncCommitteeJob;
+  // The signature's slot is for the current slot
+  if (chain.clock.currentSlot < syncCommittee.slot) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.FUTURE_SLOT,
+      currentSlot: chain.clock.currentSlot,
+      syncCommitteeSlot: syncCommittee.slot,
+      job: syncCommitteeJob,
+    });
+  }
+
+  if (chain.clock.currentSlot > syncCommittee.slot) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.PAST_SLOT,
+      currentSlot: chain.clock.currentSlot,
+      syncCommitteeSlot: syncCommittee.slot,
+      job: syncCommitteeJob,
+    });
+  }
+
+  // The block being signed over has been seen
+  if (!chain.forkChoice.hasBlock(syncCommittee.beaconBlockRoot)) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.UNKNOWN_BEACON_BLOCK_ROOT,
+      beaconBlockRoot: syncCommittee.beaconBlockRoot as Uint8Array,
+      job: syncCommitteeJob,
+    });
+  }
+
+  // There has been no other valid sync committee signature with same slot and validatorIndex
+  if (db.seenSyncCommiteeCache.hasSyncCommitteeSignature(syncCommittee)) {
```

Comment:
If we already have a validated sync committee object that includes this one, then this one is also valid right? should we then return early and accept it instead of throwing?

Reply:
we throw here and the gossip validation would cache it then it'll throw `IGNORE` error, the spec said to throw `IGNORE` error in this case instead of an accept